src/journal-remote/journal-upload.c

```diff
@@ -520,9 +526,53 @@ static int perform_upload(Uploader *u) {
                 return log_error_errno(SYNTHETIC_ERRNO(EIO),
                                        "Upload to %s finished with unexpected code %ld: %s",
                                        u->url, status, strna(u->answer));
-        else
+        else {
+                if (u->compression->algorithm == COMPRESSION_NONE) {
+                        Compression algorithm;
+                        bool found = false;
+                        hcode = curl_easy_header(u->easy, "Accept-Encoding", 0, CURLH_HEADER, -1, &encoding);
+                        if (hcode == CURLHE_OK && encoding && encoding->value) {
+                                while(encoding->value != NULL && !found) {
+                                        _cleanup_free_ char *encVal = NULL;
+                                        _cleanup_free_ char *a = NULL;
+
+                                        if (extract_first_word((const char **)&encoding->value, &encVal, ",", 0) < 0)
+                                                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Failed to extract Accept-Encoding value");
+                                        char *t = strchr(encVal, ';');
+                                        if (t) {
+                                                a = strndup(encVal, t-encVal);
+                                                if (!a)
+                                                        return log_error_errno(SYNTHETIC_ERRNO(ENOMEM), "Failed to extract encoding from Accept-Encoding header");
+                                        } else
+                                                a = strdup(encVal);
+                                        algorithm = compression_lowercase_from_string(a);
+                                        STRV_FOREACH(c, arg_compression)
```

Comment:
humm, STRV_FOREACH is for NULL terminated char** array. do not use for anything else